### PR TITLE
Visual testing capabilities version 2

### PR DESCRIPTION
**Pull request by nadvolod**
These are visual testing capabilities as we discussed:

Visual testing is enabled like this:
`SauceOptions sauceOptions = new SauceOptions().visual();`

Configuration:
```
sauceOptions.setName("testName");
sauceOptions.setViewportSize("1280x1024");
```

**What I don't like:**
- The user can set visual capabilities without actually running a visual test. Not necessarily bad I think. But is a confusing API. 
In one instance I chose to throw an `OptionForVisualTestingOnlyException` exception if `visualTesting` isn't true. But that's also awkward for a few reasons. Probably want to get away from this, but I included it as an option.
- I don't like adding if/else blocks to the currently working methods to perform different operations if `visualTesting` is enabled. First, that can break the original functionality. Second, it's making the code messy. I think I can resolve this in a cleaner manner using the Strategy pattern as I did [here](https://github.com/saucelabs/sauce_bindings/pull/218).

### CODE REVIEW

**Comment by nadvolod** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- java/src/test/java/com/saucelabs/saucebindings/integration/VisualTests.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/SauceLabs/sauce_bindings/pullRequest?prid=6379142&bid=20693430)

File: java/src/test/java/com/saucelabs/saucebindings/integration/VisualTests.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+import com.saucelabs.saucebindings.SauceOptions;
+import com.saucelabs.saucebindings.SauceSession;
+import org.junit.Test;
+import org.openqa.selenium.remote.RemoteWebDriver;
+
+import static org.junit.Assert.assertNotNull;
+
+public class VisualTests {
+    private RemoteWebDriver webDriver;
```
Comment (nadvolod) [inline java/src/test/java/com/saucelabs/saucebindings/integration/VisualTests.java:11]: Codacy found an issue: [Perhaps 'webDriver' could be replaced by a local variable.](https://app.codacy.com/manual/SauceLabs/sauce_bindings/pullRequest?prid=6379142)

**Comment by nadvolod** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- java/src/test/java/com/saucelabs/saucebindings/integration/VisualTests.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/SauceLabs/sauce_bindings/pullRequest?prid=6379142&bid=20693430)

File: java/src/test/java/com/saucelabs/saucebindings/integration/VisualTests.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+import com.saucelabs.saucebindings.SauceOptions;
+import com.saucelabs.saucebindings.SauceSession;
+import org.junit.After;
+import org.junit.Test;
+import org.openqa.selenium.remote.RemoteWebDriver;
+
+import static org.junit.Assert.assertNotNull;
+
+public class VisualTests {
+    private RemoteWebDriver webDriver;
```
Comment (nadvolod) [inline java/src/test/java/com/saucelabs/saucebindings/integration/VisualTests.java:13]: Codacy found an issue: [Perhaps 'webDriver' could be replaced by a local variable.](https://app.codacy.com/manual/SauceLabs/sauce_bindings/pullRequest?prid=6379142)

**Comment by nadvolod** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- java/src/test/java/com/saucelabs/saucebindings/integration/VisualTests.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/SauceLabs/sauce_bindings/pullRequest?prid=6379142&bid=20693430)

**Comment by nadvolod** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- java/src/test/java/com/saucelabs/saucebindings/integration/VisualTests.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/SauceLabs/sauce_bindings/pullRequest?prid=6379142&bid=20693430)

**Comment by nadvolod** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- java/src/test/java/com/saucelabs/saucebindings/integration/VisualTests.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/SauceLabs/sauce_bindings/pullRequest?prid=6379142&bid=20693430)

**Comment by nadvolod** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- java/src/test/java/com/saucelabs/saucebindings/integration/VisualTests.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/SauceLabs/sauce_bindings/pullRequest?prid=6379142&bid=20693430)

**Comment by titusfortner** [comment]
Superceded by #256

**Pull request closed without merging**